draft-ietf-quic-transport.md

```diff
@@ -3892,6 +3889,7 @@ errors are detected.
 Endpoints validate ECN for packets sent on each network path independently.  An
 endpoint thus validates ECN on new connection establishment, when switching to a
 server's preferred address, and on active connection migration to a new path.
+If validation fails, an endpoint could also periodically attempt validation.
```

Comment:
Why is this a could. Ignoring ECN is inappropriate and dangerous to the health of other flows. (If a flow received CE-marks it could reasonably assume ECN is working, if it see nothing it needs at some point to recheck this works.)

I'd expect this to be needed: Why therefore is this not a SHOULD?

Reply:
An endpoint receiving CE marks should always feed them back to the sender. Validation is about actively marking packets as ECT by the sender. However, I think it would be nice to be more explicit and add a reference to the respective section at least.

Reply:
This is a could and not a SHOULD because making a recommendation would require us to make that recommendation actionable.  That means describing the circumstances under which revalidation is desirable.

I realize that not all routes are static, but you would have to establish a case for revalidation in order to make any sort of statement with normative force.

Reply:
To be clear: I suggest we keep the text that says: "An endpoint validates the use of ECN on the path, both during connection establishment and when migrating to a new path ({{migration}})." 

... Although the simplest way to do this is to monitor an increase in the ECN counters reported.
 
I suspect there have been too many threads/issues related to ECN text (some have diverged and some concluded with no action) and those people (me included)  involved should now read a clean version of the ECN text. I do not expect this text to be hard to finalise.

Reply:
This appears immediately above:

> It is possible for faulty network devices to corrupt or erroneously drop packets with ECN markings.  To provide robust connectivity in the presence of such devices, each endpoint independently validates ECN counts and disables ECN if errors are detected.

This would seem to suffice.